lib/src/media/stream/client.dart

```diff
@@ -5,51 +5,112 @@ import 'package:flutter_webrtc/flutter_webrtc.dart';
 import '../../domain/web_rtc/web_rtc_client/web_rtc_client.dart';
 import '../../gen/proto/stream/v1/stream.pbgrpc.dart';
 
-class StreamClient {
+class StreamManager {
+  final Map<String, MediaStream> _streams = {};
+  final Map<String, StreamClient> _clients = {};
   final WebRtcClientChannel _channel;
   final StreamServiceClient _client;
-  // ignore: close_sinks
-  final StreamController<MediaStream> _streamController = StreamController<MediaStream>.broadcast();
-  MediaStream? _stream;
   StreamSubscription? _errorHandler;
 
-  StreamClient(this._channel) : _client = StreamServiceClient(_channel) {
+  static final Finalizer<StreamSubscription?> _finalizer = Finalizer((p0) {
+    p0?.cancel();
+  });
+
+  StreamManager(this._channel) : _client = StreamServiceClient(_channel) {
+    _finalizer.attach(this, _errorHandler);
     _channel.rtcPeerConnection.onAddStream = (MediaStream stream) {
-      _stream = stream;
-      _streamController.add(stream);
+      addStream(stream);
     };
 
     _channel.rtcPeerConnection.onConnectionState = (state) {
       if (state == RTCPeerConnectionState.RTCPeerConnectionStateFailed ||
           state == RTCPeerConnectionState.RTCPeerConnectionStateDisconnected) {
         _errorHandler = Stream.periodic(const Duration(seconds: 1)).listen((_) {
```

Comment:
[nit] i don't think you introduced this, so we can address later. we need to cancel this StreamSubscription at some point.

Reply:
This should be canceled in the `Finalizer` I introduced